custom_components/smarthashtag/sensor.py

```
@@ -1454,6 +1487,29 @@ def __init__(
+            key = remove_vin_from_key(self.entity_description.key)
+            vin = vin_from_key(self.entity_description.key)
+            if key == "odometer":
+                data = getattr(
+                    self.coordinator.account.vehicles.get(vin),
+                    key,
+                )
+                try:
+                    if isinstance(data, str):
+                        float(data)
+                except AttributeError:
+                    raise ValueError("Value is not a number")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)